STYLE.md

```diff
@@ -46,6 +35,36 @@
 * There are probably a few other things missing from this list. We will add them as they
   are brought to our attention.
 
+# Error handling
+
+A few general notes on our error handling philosophy:
+
+* All error code returns should be checked.
+* At a very high level, our philosophy is that errors that are *likely* to happen should be
+  gracefully handled. Examples of likely errors include any type of network error, bad data
+  returned by an API call, bad data read from runtime files, etc. Errors that are *unlikely* to
+  happen should lead to process death, under the assumption that the additional burden of defensive
+  coding and testing is not an effective use of time for an error that should not happen given
+  proper system setup. Examples of these types of errors include not being able to open the shared
+  memory region, an invalid initial JSON config read from disk, system calls that should not fail
+  assuming correct parameters (which should be validated via tests), etc.
+* OOM events (both memory and FDs) are considered fatal crashing errors. This rule is again based
+  on the philosophy that the engineering costs of properly handling these cases is not worth it.
+  Time is better spent designing proper system controls that shed load if resource usage becomes
+  too high, etc.
```

Comment:
I think part of this philosophy is also that restarts are fast, reliable and cheap. Can you add something to this effect? This won't be universally true, so it'd help to have it as a strawman.